### PR TITLE
Added KL Divergence loss tutorial

### PR DESCRIPTION
**Pull request by thomelane**
A lot of confusion has been caused over how to use KL Divergence loss function. We need to document it, and this tutorial is an extension of @NRauschmayr's tutorial on loss functions, from which this tutorial will be linked. Shows difference between `from_logits` as True and False, and discussed various cases where divergence calculated could be wrong/different from true calculation.

### CODE REVIEW

**Comment by thomelane** [comment]
@NRauschmayr please review, thanks!

File: python/guide/packages/gluon/loss/kl_divergence.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Kullback-Leibler (KL) Divergence
+
+Kullback-Leibler (KL) Divergence is a measure of how one probability distribution is different from a second, reference probability distribution. Smaller KL Divergence values indicate more similar distributions and, since this loss function is differentiable, we can use gradient descent to minimize the KL divergence between network outputs and some target distribution. As an example, this can be used in Variational Autoencoders (VAEs), and reinforcement learning policy networks such as [Trust Region Policy Optimization (TRPO)](https://arxiv.org/abs/1502.05477).
+
+In MXNet Gluon, we can use [`KLDivLoss`](https://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.loss.KLDivLoss.html) to compare categorical distributions. One important thing to note is that the KL Divergence is an asymmetric measure: order matters and we should compare our predicted distribution with our target distribution in that order. Another thing to note is that there are two ways to use [`KLDivLoss`](https://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.loss.KLDivLoss.html) that depend on how we set `from_logits` (which has a default value of true).
```
Comment (NRauschmayr) [inline python/guide/packages/gluon/loss/kl_divergence.md]: I would add to the statement about asymmetry: KL(P|Q) is not the same like KL(Q|P) though both measure the similarity between P and Q.

File: python/guide/packages/gluon/loss/kl_divergence.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    # log the distribution
+    pred_batch = pred_batch.log()
+    # create loss (assuming we have a logged prediction distribution)
+    loss_fn = mx.gluon.loss.KLDivLoss(from_logits=True)
+    divergence = loss_fn(pred_batch, target_batch)
+    return divergence.asscalar()
+```
+
+```
+print("Distribution 1 compared with Distribution 2: {}".format(kl_divergence(dist_1, dist_2)))
```
Comment (NRauschmayr) [inline python/guide/packages/gluon/loss/kl_divergence.md]: PEP8: line too long

File: python/guide/packages/gluon/loss/kl_divergence.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    pred_batch = mx.nd.array(dist_a).expand_dims(0)
+    target_batch = mx.nd.array(dist_b).expand_dims(0)
+    # create loss (assuming we have a logged prediction distribution)
+    loss_fn = mx.gluon.loss.KLDivLoss(from_logits=False)
+    divergence = loss_fn(pred_batch, target_batch)
+    return divergence.asscalar()
+```
+
+```
+print("Distribution 1 compared with Distribution 2: {}".format(kl_divergence_not_from_logits(output, dist_2)))
```
Comment (NRauschmayr) [inline python/guide/packages/gluon/loss/kl_divergence.md]: PEP8

File: python/guide/packages/gluon/loss/kl_divergence.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+Occasionally, you might have issues with [`KLDivLoss`](https://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.loss.KLDivLoss.html). One common issue arises when the support of the distributions being compared are not the same. 'Support' here is referring to the values of the distribution which have a non-zero probability. Conveniently, all our examples above had the same support, but we might have a case where some categories have a probability of 0.
+
+
+```
+dist_4 = np.array([0, 0.9, 0, 0.1])
+```
+
+```
+print("Distribution 4 compared with Distribution 1: {}".format(kl_divergence(dist_4, dist_1)))
```
Comment (NRauschmayr) [inline python/guide/packages/gluon/loss/kl_divergence.md]: PEP8

File: python/guide/packages/gluon/loss/kl_divergence.md (hunk 1 above)
Comment (thomelane) [reply python/guide/packages/gluon/loss/kl_divergence.md]: Got a comment on it. Added `KL(P,Q) != KL(Q,P)`.

**Comment by thomelane** [comment]
thanks for the review @NRauschmayr!
